### PR TITLE
Adapt high-availability configuration of seed system components

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area high-availability
/kind enhancement

**What this PR does / why we need it**:
This PR adapts the HA configuration of the non-observability-related seed system components.
Concretely, the components are labeled according to their type and custom replica/affinity changes have been dropped.
The webhook introduced with https://github.com/gardener/gardener/pull/6967 will take care to properly inject the required HA configuration.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6529
Similar to https://github.com/gardener/gardener/pull/6989 and https://github.com/gardener/gardener/pull/6992

**Special notes for your reviewer**:
✅ ~Depends on https://github.com/gardener/gardener/pull/6967 which needs to be merged first.~

/cc @timuthy 

**Release note**:

```feature operator
All non-observability-related seed system components are now running with configuration for high-availability according to [the conventions](https://github.com/gardener/gardener/blob/master/docs/development/high-availability.md).
```

### CODE REVIEW

**Review by timuthy — COMMENTED** [review]
Thank you very much 🙂 
/lgtm
/approve

File: pkg/operation/botanist/component/dependencywatchdog/bootstrap.go [hunk 1]
```diff
@@ -280,8 +289,37 @@ func (b *bootstrapper) Deploy(ctx context.Context) error {
 				},
 			},
 		}
+
+		podDisruptionBudget client.Object
+		maxUnavailable      = intstr.FromInt(1)
 	)
 
+	if version.ConstraintK8sGreaterEqual121.Check(b.values.KubernetesVersion) {
```
Comment (timuthy) [inline pkg/operation/botanist/component/dependencywatchdog/bootstrap.go:297]: See https://github.com/gardener/gardener/pull/6992#discussion_r1021622137.

Comment (rfranzke) [reply pkg/operation/botanist/component/dependencywatchdog/bootstrap.go:297]: Let's merge https://github.com/gardener/gardener/pull/6992 first, then I can rebase this PR on top of it and use the new `versionutils.Normalize` function accordingly.

/approve cancel
/hold

Comment (rfranzke) [reply pkg/operation/botanist/component/dependencywatchdog/bootstrap.go:297]: OK @timuthy, now done with [1344f34](https://github.com/gardener/gardener/pull/6982/commits/1344f34c2420b8d6a4aadd065cc3bbe218c25e41).
Please check.
/unhold

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**